### PR TITLE
🐛 Fix bug with requeuing machines from AWSCluster requests

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

**What this PR does / why we need it**:
This PR fixes an issue with listing machines using a machine label.

/assign @vincepri

### CODE REVIEW

File: controllers/awsmachine_controller.go [hunk 1]
```diff
@@ -415,13 +415,13 @@ func (r *AWSMachineReconciler) AWSClusterToAWSMachines(o handler.MapObject) []ct
 	}
 
 	labels := map[string]string{clusterv1.MachineClusterLabelName: c.Name}
```
Comment (vincepri) [inline controllers/awsmachine_controller.go]: Doesn't this assume that the AWSCluster is going to have the same name as the Cluster?

Comment (detiber) [reply controllers/awsmachine_controller.go]: This is used in the label matcher below for getting the machines for the cluster in the query below.

File: controllers/awsmachine_controller.go [hunk 2]
```diff
@@ -415,13 +415,13 @@ func (r *AWSMachineReconciler) AWSClusterToAWSMachines(o handler.MapObject) []ct
...
-	machineList := &infrav1.AWSMachineList{}
+	machineList := &clusterv1.MachineList{}
 	if err := r.List(context.Background(), machineList, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
-		r.Log.Error(err, "failed to list AWSMachines", "AWSCluster", c.Name, "Namespace", c.Namespace)
+		r.Log.Error(err, "failed to list Machines", "AWSCluster", c.Name, "Namespace", c.Namespace)
 		return nil
 	}
 	for _, m := range machineList.Items {
-		name := client.ObjectKey{Namespace: m.Namespace, Name: m.Name}
+		name := client.ObjectKey{Namespace: m.Namespace, Name: 	m.Spec.InfrastructureRef.Name}
```
Comment (detiber) [inline controllers/awsmachine_controller.go]: I'm wondering if we need to guard against the InfrastructureRef not being set yet here.

Comment (chuckha) [reply controllers/awsmachine_controller.go]: I thought about that yesterday after i turned my computer off. +1, will update

File: controllers/awsmachine_controller.go (hunk 1 above)
Comment (detiber) [reply controllers/awsmachine_controller.go]: nm, @vincepri is correct, we are using the name of the AWSCluster here as the Cluster name.

**Pull request merged**